aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -377,7 +376,8 @@ Tensor logsumexp(const Tensor &self, int64_t dim_, bool keepdim) {
 
 // MULTI DIM REDUCE ###########################################################
 
-template <Tensor (reduce_1)(const Tensor &, int64_t, bool)>
+template <Tensor (reduce_1)(const Tensor &, int64_t, bool),
+    Tensor& (reduce_1_out)(Tensor& result, const Tensor &, int64_t, bool)>
```

Comment:
OK, after spending 10min trying to keep in my head exactly how the intermediate buffers are being written to, I think this deserves some sort of comment + diagram saying how it works on some simple example :)

Reply:
Thanks for reviewing! Will do!